tests/test_other.py

```diff
@@ -136,6 +138,20 @@ def do_other_test(self, dirname, emcc_args=[], run_args=[]):
     seen = run_js('a.out.js', args=run_args, stderr=PIPE, full_output=True) + '\n'
     self.assertContained(expected, seen)
 
+  def do_smart_test(self, source, name='test.cpp', literals=[], regexes=[],
```

Comment:
I worry a little about adding even more ways to compile and run test code.  I can't think of a suitable replacement right now.

Can you at least move it up next to `do_other_test` and document it?

Reply:
It is right next to `do_other_test`?